src/main/java/org/alfresco/heartbeat/HeartBeat.java

```diff
@@ -120,7 +127,18 @@ public HeartBeat(final ApplicationContext context, final Boolean testMode)
             }
             this.licenseService = licenseService;
 
-            // We force the job to be scheduled regardless of the potential state of the licenses
+            JobLockService jobLockService = null;
+            try
+            {
+                jobLockService = (JobLockService) context.getBean("jobLockService");
+            }
+            catch (NoSuchBeanDefinitionException e)
+            {
+                logger.error("jobLockService not found", e);

```

Comment:
Is it not possible to inject the jobLockService?
This bean, if not found, is a problem that should prevent the system from starting at all.
As you have it, there will be an error logged followed by NPEs.  Just inject the service you need.